### ISSUE TITLE
[Console] HelpCommand strips arguments with same names as output styles

### ISSUE THREAD

**Issue by pbull** [symfony/symfony#24225]
| Q                | A
| ---------------- | -----
| Bug report?      | yes
| Feature request? | no
| BC Break report? | no
| RFC?             | no
| Symfony version  | 2.7.x, 3.3.x

When a console command defines one or more input arguments that have the same name(s) as defined OutputFormatterStyles, the HelpCommand output strips those arguments from the Usage section of the output and applies that style to the content that follows.

This is occurring because in `Symfony\Component\Console\Input\InputDefinition::getSynopsis()` (https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Console/Input/InputDefinition.php#L398) those argument names are wrapped in `<` and `>` and so the argument is treated as the opening tag of a style wrapper:

```
        foreach ($this->getArguments() as $argument) {
            $element = '<'.$argument->getName().'>';
[...]
        }
```

An example command that defines the argument names "info", "comment", and "foo" as follows:

```
    protected function configure() {
        $this->addArgument('info', InputArgument::REQUIRED, 'The first argument.');
        $this->addArgument('comment', InputArgument::OPTIONAL, 'The second argument.');
        $this->addArgument('foo', InputArgument::OPTIONAL, 'The third argument.');
    }
```
... would appear as follows when running `help` for that command. Note that the yellow foreground color of the default `comment` style is being applied to the following text in the output:

![console1](https://user-images.githubusercontent.com/580026/30499296-e9f187ba-9a27-11e7-9b49-b7aa0db9a541.png)

... when the expected output would be more like:

![console2](https://user-images.githubusercontent.com/580026/30499356-2c8f172c-9a28-11e7-9cfd-38ac87ae620a.png)

**Comment by pbull** [comment]
While a simple fix is to prepend the first `<` with a backslash (and this fixes both the text and markdown output of the help command), this will require updating a number of tests to account for that backslash, and that character is also inserted into the JSON and XML formatted output from help. I don't know if that is acceptable, or if the `JsonDescriptor` and `XmlDescriptor` should account for that somehow (e.g. passing through `stripslashes()`?):

```
--- a/src/Symfony/Component/Console/Input/InputDefinition.php
+++ b/src/Symfony/Component/Console/Input/InputDefinition.php
@@ -395,7 +395,7 @@ class InputDefinition
         }

         foreach ($this->getArguments() as $argument) {
-            $element = '<'.$argument->getName().'>';
+            $element = '\<'.$argument->getName().'>';
             if (!$argument->isRequired()) {
                 $element = '['.$element.']';
             } elseif ($argument->isArray()) {
```

**Comment by ro0NL** [comment]
using `\` here looks like a leaked implementation detail, cant we simply escape on the rendering side?

**Comment by pbull** [comment]
Yes, looks like this could be handled more gracefully in the `TextDescriptor` and `MarkdownDescriptor`. I'll give that a try.

**Comment by sroze** [comment]
PRed a fix in #24455
